adds Amazon Games, latest version

rdmartins opened the pull request:
Adds manifest for Amazon Games app, always the
latest version of the client

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/20235)

ItzLevvie commented on manifests/a/Amazon/Games/latest/Amazon.Games.locale.en-US.yaml:
```diff
@@ -0,0 +1,26 @@
+PackageIdentifier: Amazon.Games
+PackageVersion: latest
+PackageLocale: en-US
+Publisher: Amazon.com, Inc.
```
```suggestion
Publisher: Amazon.com Services, Inc.
```

rdmartins replied:
I can change, but it's not from Amazon Services. I got the information from the app about tab.

ItzLevvie replied:
Publisher needs to match Control Panel because Windows Package Manager uses this information for `winget list` and `winget upgrade`.

rdmartins replied:
Good to know, I'll take this information from the control panel for the next manifests

ItzLevvie commented on manifests/a/Amazon/Games/latest/Amazon.Games.locale.en-US.yaml:
```diff
@@ -0,0 +1,26 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultlocale.1.0.0.schema.json
+
+PackageIdentifier: Amazon.Games
+PackageVersion: latest
```
```suggestion
PackageVersion: 1.7.3024.2
```

rdmartins replied:
The link always point to the latest version of the app.
When Amazon release a new version, the same link will install a newer version.

ItzLevvie replied:
PackageVersion needs to match Control Panel because Windows Package Manager uses this information for `winget list` and `winget upgrade`.

ItzLevvie replied:
wingetbot will automatically update the version when there's a new release.

rdmartins replied:
Oh, good to know!

This information needs to match for the app to be uninstalled successfully?

ItzLevvie commented on manifests/a/Amazon/Games/latest/Amazon.Games.installer.yaml:
```diff
@@ -0,0 +1,20 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.0.0.schema.json
+
+PackageIdentifier: Amazon.Games
+PackageVersion: latest
```
```suggestion
PackageVersion: 1.7.3024.2
```

rdmartins replied:
The link always point to the latest version of the app.
When Amazon release a new version, the same link will install a newer version.

ItzLevvie replied:
PackageVersion needs to match Control Panel because Windows Package Manager uses this information for `winget list` and `winget upgrade`.

ItzLevvie replied:
wingetbot will automatically update the version when there's a new release.

ItzLevvie commented on manifests/a/Amazon/Games/latest/Amazon.Games.yaml:
```diff
@@ -0,0 +1,8 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.0.0.schema.json
+
+PackageIdentifier: Amazon.Games
+PackageVersion: latest
```
```suggestion
PackageVersion: 1.7.3024.2
```

ItzLevvie replied:
PackageVersion needs to match Control Panel because Windows Package Manager uses this information for `winget list` and `winget upgrade`.

ItzLevvie replied:
wingetbot will automatically update the version when there's a new release.

ItzLevvie commented on manifests/a/Amazon/Games/latest/Amazon.Games.locale.en-US.yaml:
```diff
@@ -0,0 +1,26 @@
+LicenseUrl: https://www.amazon.com/gp/help/customer/display.html?ie=UTF8&nodeId=508088
+Copyright: 2019-2021, Amazon.com, Inc. or its affiliates
+CopyrightUrl: https://www.amazon.com/gp/help/customer/display.html?ie=UTF8&nodeId=508088
+ShortDescription: Amazon Games App
```
```
ShortDescription: Amazon Games App
```
Please write a better description.

rdmartins replied:
The name of the software is "Amazon Games App",

ItzLevvie replied:
e.g. `Amazon Games app is a Windows PC game launcher that allows you to claim, download, and play new and classic Windows PC games every month from Prime Gaming`

rdmartins replied:
I'll keep that in mind for my future manifests

rdmartins committed:
adds Amazon Games, latest version

ItzLevvie reviewed (changes requested):
Please see latest review.

rdmartins committed:
Update manifests/a/Amazon/Games/latest/Amazon.Games.locale.en-US.yaml

Co-authored-by: Levvie - she/her <11600822+ItzLevvie@users.noreply.github.com>

rdmartins committed:
Update manifests/a/Amazon/Games/latest/Amazon.Games.locale.en-US.yaml

Co-authored-by: Levvie - she/her <11600822+ItzLevvie@users.noreply.github.com>

rdmartins committed:
Update manifests/a/Amazon/Games/latest/Amazon.Games.installer.yaml

Co-authored-by: Levvie - she/her <11600822+ItzLevvie@users.noreply.github.com>

ItzLevvie commented on manifests/a/Amazon/Games/latest/Amazon.Games.locale.en-US.yaml:
```diff
@@ -0,0 +1,26 @@
+LicenseUrl: https://www.amazon.com/gp/help/customer/display.html?ie=UTF8&nodeId=508088
+Copyright: 2019-2021, Amazon.com, Inc. or its affiliates
+CopyrightUrl: https://www.amazon.com/gp/help/customer/display.html?ie=UTF8&nodeId=508088
+ShortDescription: Amazon Games App
```
```suggestion
ShortDescription: Amazon Games app is a Windows PC game launcher that allows you to claim, download, and play new and classic Windows PC games every month from Prime Gaming.
```

rdmartins committed:
Update manifests/a/Amazon/Games/latest/Amazon.Games.yaml

Co-authored-by: Levvie - she/her <11600822+ItzLevvie@users.noreply.github.com>

ItzLevvie reviewed (changes requested):
You'll have to update the version folder as well.

rdmartins committed:
Update manifests/a/Amazon/Games/latest/Amazon.Games.locale.en-US.yaml

Co-authored-by: Levvie - she/her <11600822+ItzLevvie@users.noreply.github.com>

rdmartins commented:
> 
> 
> You'll have to update the version folder as well.

Just did that.

Where can I find help on how to use the application product code?

jedieaston commented:
There's lots of different ways to get it, since this is an exe your best bet is probably `winget list` or looking in the registry directly in either:

`HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall`

or for 32-bit software:

`HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Microsoft\Windows\CurrentVersion\Uninstall`.

(it isn't commonly needed for exe manifests though, it's mostly for MSI/wix).

jedieaston commented:
<img width="329" alt="image" src="https://user-images.githubusercontent.com/21368066/125005866-d1521d80-e02a-11eb-8628-38bd6d529e9d.png">

🤔

Out of curiosity, what time zone are you in @rdmartins?

rdmartins commented:
I'm on BRT, but my Windows clock is messed up: I have dual boot and keep my Windows in UTC, but now I updated to Windows 11 and my configuration was lost. I fixed in the registry, just need to reboot to fix it.

rdmartins committed:
adds product code

Now uninstall works

rdmartins committed:
fixes installer version

ItzLevvie reviewed (approved):
![image](https://user-images.githubusercontent.com/11600822/125071757-a363ec00-e0b1-11eb-8d66-b3795b08a5e4.png)
